src/sacn/source.c

```diff
@@ -29,264 +50,474 @@
+ * to be sent will have its force_synchronization option flag set.
  *
- * This behavior is rarely necessary, but provided for unanticipated use cases and debug use. Use
- * this function to send a sACN packet for a universe immediately (i.e. in between calls to
- * sacn_process_sources()).
+ * If no synchronization universe is configured, this function acts like a direct call to sacn_source_set_dirty().
  *
- * Note that calling this function could result in behavior which violates E1.31-2016 &sect; 6.6.
+ * TODO: At this time, synchronization is not supported by this library.
  *
- * \param[in] start_codes Array of source/start code pairs on which to send data immediately.
- * \param[in] num_start_codes Size of the start_codes array.
+ * \param[in] handle Handle to the source to mark as dirty.
+ * \param[in] universe Universe to mark as dirty.
  */
-void sacn_sources_send_now(const SacnSourceStartCodePair* start_codes, size_t num_start_codes)
+void sacn_source_set_dirty_and_force_sync(sacn_source_t handle, uint16_t universe)
 {
-  ETCPAL_UNUSED_ARG(start_codes);
-  ETCPAL_UNUSED_ARG(num_start_codes);
   // TODO
+
+  ETCPAL_UNUSED_ARG(handle);
+  ETCPAL_UNUSED_ARG(universe);
 }
 
 /*!
- * \brief Process created sources and do the actual sending of sACN data.
+ * \brief Process created sources and do the actual sending of sACN data on all universes.
+ *
+ * Note: Unless you created the source with manually_process_source set to true, this will be automatically
+ * called by an internal thread of the module. Otherwise, this must be called at the maximum rate
+ * at which the application will send sACN.
  *
- * Must be called at the maximum rate at which the application will send sACN. Sends data for
- * start codes which have been marked dirty, and sends keep-alive data for start codes which are
- * static. Also destroys sources that have been marked for termination after sending the required
+ * Sends data for universes which have been marked dirty, and sends keep-alive data for universes which are
+ * haven't changed. Also destroys sources & universes that have been marked for termination after sending the required
  * three terminated packets.
  *
  * \return Current number of sources tracked by the library. This can be useful on shutdown to
  *         track when destroyed sources have finished sending the terminated packets and actually
  *         been destroyed.
  */
-size_t sacn_process_sources(void)
+size_t sacn_source_process_sources(void)
 {
-  // TODO
+  // TODO CHRISTIAN
   return 0;
 }
+
+/*!
+ * \brief Resets the underlying network sockets for the sACN source..
+ *
+ * This is typically used when the application detects that the list of networking interfaces has changed.
+ *
+ * After this call completes successfully, the all universes on a source are considered new & dirty by the
+ * per-address priority logic in sacn_source_process_sources.
```

Comment:
Not sure why PAP is mentioned here.

Reply:
I was trying to get across the fact that calling this would execute any "take-control" logic that is required by the per-address support.  We can tweak this after implementation to have a better idea what that would be.

Reply:
It's in the per-address priorities spec.

Reply:
What I'm getting at is, does someone outside ETC who is reading this documentation have the tools to understand what this means?

Reply:
I've changed the text to
 * After this call completes successfully, all universes on a source are considered to be dirty and have
 * new values and priorities. It's as if the source just started sending values on that universe.

What do you think?

Reply:
Yeah, I like that.

Reply:
Sounds good to me.